src/lib/yang/alarms.lua

```diff
@@ -43,6 +43,10 @@ function notifications ()
    local ret = {}
    local notifications = state.notifications
    for k,v in pairs(notifications.alarm) do
+      -- TODO: Improve JSON encode / decode to support nested tables
+      if type(v.alt_resource) == 'table' then
+         v.alt_resource = table.concat(v.alt_resource, ',')
+      end
```

Comment:
I'm not sure about this change. In one hand, the encode/decode code for alt-resource serialization seems OK to me. It works with a native Lua table. But here `alt_resource` is encoded to a string of separated by commas, but only for notifications.

An `alarm-notification` is defined in the alarms schema as:

```
notification alarm-notification {
        uses common-alarm-parameters;
        uses alarm-state-change-parameters;
}
```

Basically has the same attributes and structure as an alarm. If things are working ok when ptree deserializes notifications to a json object it should deserialize all attributes (https://github.com/Igalia/snabb/blob/lwaftr/src/lib/ptree/ptree.lua#L234).